src/features/workspace/components/__tests__/workspace-preferred-model.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { render } from "@/lib/test-utils";
+import { screen, waitFor } from "@testing-library/react";
+import { WorkspacePreferredModel } from "../workspace-preferred-model";
+import userEvent from "@testing-library/user-event";
+
+test("render model overrides", () => {
+  render(
+    <WorkspacePreferredModel
+      isArchived={false}
+      workspaceName="fake-workspace"
+    />,
+  );
+  expect(screen.getByText(/preferred model/i)).toBeVisible();
+  expect(
+    screen.getByText(
+      /select the model you would like to use in this workspace./i,
+    ),
+  ).toBeVisible();
+  expect(
+    screen.getByRole("button", { name: /select the model/i }),
+  ).toBeVisible();
+  expect(screen.getByRole("button", { name: /save/i })).toBeVisible();
+});
+
+test("submit preferred model", async () => {
+  render(
+    <WorkspacePreferredModel
+      isArchived={false}
+      workspaceName="fake-workspace"
+    />,
+  );
+
+  await userEvent.click(
+    screen.getByRole("button", { name: /select the model/i }),
+  );
+
+  await userEvent.click(
+    screen.getByRole("option", {
+      name: "anthropic/claude-3.5",
+    }),
+  );
+
+  await userEvent.click(screen.getByRole("button", { name: /save/i }));
+
+  await waitFor(() => {
+    expect(
+      screen.getByText(
+        /preferred model on fake-workspace successfully submitted!/i,
```

Comment:
The corresponding tweak for my other comment

```suggestion
        /preferred model for "fake-workspace" updated/i
```

Reply:
yep makes sense, 🙏